### PR TITLE
Also migrate the "useFolderUrl" setting

### PR DESCRIPTION
**Pull request by fritzmg**
| Q                | A
| -----------------| ---
| Fixed issues     | -
| Docs PR or issue | -

When upgrading a Contao 4.9 instance to 4.10 I noticed that the `folderUrl` setting wasn't migrated to the `tl_page` settings.

_Note:_ I need some help with the unit tests, as I am not sure how to properly mock the `Config` singleton, which also has static methods.

### CODE REVIEW

**Comment by m-vo** [comment]
> Note: I need some help with the unit tests, as I am not sure how to properly mock the Config singleton, which also has static methods.

What do you need exactly? Maybe you can solve it with `$this->mockClassWithProperties(Config::class, [...]);`

**Comment by fritzmg** [comment]
> What do you need exactly? Maybe you can solve it with `$this->mockClassWithProperties(Config::class, [...]);`

Well the static function is not a property. I want to do 

```php
$config = $this->createMock(Config::class);
$config
    ->expects($this->once())
    ->method('has')
    ->with('folderUrl')
    ->willReturn(false)
;
``` 

But this will fail:

```
1) Contao\CoreBundle\Tests\Migration\Version410\FolderUrlMigrationTest::testDoesNothingIfFolderUrlNotEnabled
Static method "has" cannot be invoked on mock object
```

**Review by aschempp — CHANGES_REQUESTED** [review]
I don't think this will work. If `folderUrl` is disable in the root pages, the migration will always run again. The correct migration would be to check if the field exists in the database, and only migrate if it does not. That obviously won't fix current RCs, but we don't provide such fixes for RCs.

**Comment by fritzmg** [comment]
> I don't think this will work. If `folderUrl` is disable in the root pages, the migration will always run again.

Only if you re-add the setting in your `localconfig.php` or don't remove `contao.localconfig.folderUrl: true` from your `config.yml`.

But generally I agree, the migration should only run, if the database field does not exist yet.

**Comment by aschempp** [comment]
It could even be added to the `RoutingMigration` in my view 😊

**Comment by fritzmg** [comment]
Alright, I have integrated it into the `RoutingMigration` now. Which caused me to notice, that the `RoutingMigration` wasn't actually registered as a service and thus would have never been executed in the first place 🙈 

~~(also, there are no tests what so ever for any 410 migration 😶)~~

File: core-bundle/src/Migration/Version410/RoutingMigration.php [hunk 1]
```diff
@@ -39,9 +46,10 @@ class RoutingMigration extends AbstractMigration
      */
     private $prependLocale;
 
-    public function __construct(Connection $connection, string $urlSuffix = '.html', bool $prependLocale = false)
+    public function __construct(Connection $connection, ContaoFramework $framework, string $urlSuffix = '.html', bool $prependLocale = false)
```
Comment (leofeyer) [inline core-bundle/src/Migration/Version410/RoutingMigration.php:49]: You also have to adjust the `RoutingMigrationTest` class (see the CI chain).

**Comment by fritzmg** [comment]
Right, forgot about the functional tests. Updated.

**Review by aschempp — APPROVED** [review]

**Comment by leofeyer** [comment]
Thank you @fritzmg.

**Pull request merged**